internal/backend/fs.go

```diff
@@ -134,11 +134,13 @@ func (s *storageFS) CreateObject(obj Object) (Object, error) {
 	if err != nil {
 		return Object{}, err
 	}
-	encoded, err := json.Marshal(obj)
+	file, err := os.OpenFile(filepath.Join(s.rootDir, url.PathEscape(obj.BucketName), url.PathEscape(obj.Name)), os.O_CREATE|os.O_WRONLY, 0o600)
 	if err != nil {
 		return Object{}, err
 	}
-	return obj, ioutil.WriteFile(filepath.Join(s.rootDir, url.PathEscape(obj.BucketName), url.PathEscape(obj.Name)), encoded, 0o600)
+	defer file.Close()
+	err = json.NewEncoder(file).Encode(obj)
+	return obj, err
```

Comment:
I assume that for encoding this doesn't matter? Agree it's a better pattern though.

Reply:
Sorry I don't understand -- this should have the same encoding as `json.Marshal`?